js/modules/k6/execution/execution.go

```diff
@@ -297,72 +296,66 @@ func optionsAsObject(rt *goja.Runtime, options lib.Options) (*goja.Object, error
 }
 
 type tagsDynamicObject struct {
-	Runtime *goja.Runtime
-	State   *lib.State
+	runtime *goja.Runtime
+	state   *lib.State
 }
 
 // Get a property value for the key. May return nil if the property does not exist.
 func (o *tagsDynamicObject) Get(key string) goja.Value {
-	tag, ok := o.State.Tags.GetCurrentValues().Get(key)
-	if !ok {
-		return nil
+	tcv := o.state.Tags.GetCurrentValues()
+	if tag, ok := tcv.Get(key); ok {
+		return o.runtime.ToValue(tag)
 	}
-	return o.Runtime.ToValue(tag)
+	return nil
 }
 
-// Set a property value for the key. It returns true if succeed.
-// String, Boolean and Number types are implicitly converted
-// to the goja's relative string representation.
-// In any other case, if the Throw option is set then an error is raised
-// otherwise just a Warning is written.
+// Set a property value for the key. It returns true if succeed. String, Boolean
+// and Number types are implicitly converted to the goja's relative string
+// representation. Objects are checked and accepted if they contain the index
```

Comment:
Please also add a mention in the PR description that it fixes https://github.com/grafana/k6/issues/2653